### ISSUE TITLE
merging multiple .ptu files to a single hdf5

### ISSUE THREAD

**Issue by Mattyboi123** [OpenSMFS/FRETBursts#34]
Apologies if this is an already answered question or I've missed something obvious but how does one combine multiple .ptu files into a single hdf5 file prior to further analysis withe pycorrelate or fretbursts?

Best
Matt

**Comment by mayeshh** [comment]
I suggest keeping your files separate and loading the data into one large DataFrame. For example `bursts_all = pd.concat(dataframe_list, ignore_index=True)` .

**Comment by Mattyboi123** [comment]
hi thanks for the suggestion but will that preserve all of the nanotimes and still be navigable using the normal fretbursts commands?

**Comment by Mattyboi123** [comment]
by the by did you sole the issue you were having with no module named phrates_c? I now have the same issue trying to import FRETbursts on a new clean anaconda install on a new machine perhaps it is a python 3.8 issue as when  Itried to install fretbursts through conda-forge it said something about that and didn't install so  Icloned my version that was working on another machine (python 3.7) then this issue on the new machine

**Comment by mayeshh** [comment]
> hi thanks for the suggestion but will that preserve all of the nanotimes and still be navigable using the normal fretbursts commands?

Yes, each data object will have the same fields and FRETBursts will work on them in the same way.

**Comment by mayeshh** [comment]
> by the by did you sole the issue you were having with no module named phrates_c? I now have the same issue trying to import FRETbursts on a new clean anaconda install on a new machine perhaps it is a python 3.8 issue as when Itried to install fretbursts through conda-forge it said something about that and didn't install so Icloned my version that was working on another machine (python 3.7) then this issue on the new machine

I haven't upgraded to python 3.8, so I don't have experience with the compatibility. I would not expect this to be the issue, rather updates in the dependencies could cause issues (e.g. deprecations in numpy). 

Please try an “editable” installation:

```
git clone https://github.com/OpenSMFS/FRETBursts.git
cd FRETBursts
pip install -e .
```

**Comment by Mattyboi123** [comment]
Ye I did try installing like that, it didn't help. In the end I just
checked commented out the line in phrates.py that called for phrates_c.
Then another line about normpdf in matplot lib bad something else which I
forget now and it works fine on 3.8.

**Comment by mayeshh** [comment]
> Ye I did try installing like that, it didn't help. In the end I just checked commented out the line in phrates.py that called for phrates_c. Then another line about normpdf in matplot lib bad something else which I forget now and it works fine on 3.8.

Yes, the `phrates_c` module is a C optimized version of `phrates`, so it is faster but not necessary for FRETBursts to work. 

For the deprecation warning in matplotlib, I have submitted a [PR](https://github.com/OpenSMFS/FRETBursts/pull/33) for the `normpdf` issue but I haven't been able to figure out why it fails the CI checks so it has not been merged to the master branch. 

Could you print a list of your dependencies and versions for reference?

**Comment by Mattyboi123** [comment]
unfortunately, phrates_c is also required for the alex2cde filter script to
run as it uses it to implement a laplace kernel, apparently. Perhaps its a
windows not compiling things issue. I've installed visual studio and
trying with code to compile from an issue that someone else had with cython
modules not being imported into spyder

https://stackoverflow.com/questions/49837155/spyder-does-not-recognise-cython-modules-when-importing-in-a-script

**Comment by Mattyboi123** [comment]
Hello,  I still have this issue with cython scritpts not being found when runing FRETbursts even in the dedicated envoronment you suggested and after installing build tools for virtual studio. It just says 'no module named phrates_c' Can you tell me if you are running on a windows machine, how your cython extension scripts/modules are being compiled?

**Comment by Mattyboi123** [comment]
I was able to get it to work now in the manner you suggetsed - there was a path issue becuasue I was opening scripts from Spyder that were in a different folder. creating local copies within the FRETbursts dedicated environment and opening those worked perfectly with the only caveat I've yet found being that the normPPDF from matplotlib needed to be commented out - whihc as  Iunderstand it hads been worked around easily.

Thanks
